crates/codegen/syntax_templates/src/shared/cursor.rs

```diff
@@ -0,0 +1,342 @@
+use std::rc::Rc;
+
+use super::{
+    cst::{Node, RuleNode, TokenNode},
+    kinds::{RuleKind, TokenKind},
+    text_index::{TextIndex, TextRange},
+};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+struct CursorPathElement {
+    rule_node: Rc<RuleNode>,
+    child_number: usize,
+    text_offset: TextIndex,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+struct CursorLeaf {
+    node: Node,
+    child_number: usize,
+    text_offset: TextIndex,
+}
+
+impl CursorLeaf {
+    pub fn text_range(&self) -> TextRange {
```

Comment:
I wonder why are we using `text_range` here vs just `range` on nodes?

Reply:
I prefer the standardized name, so I will make that change.

In this case it's only internal because CursorLeaf isn't part of the public API.